alacritty/src/display.rs

```diff
@@ -192,20 +201,28 @@ impl Display {
         // Update OpenGL projection
         renderer.resize(&size_info);
 
-        // Clear screen
+        // We should call `clear` when window is offscreen, so when `window.show()` happens it
+        // would be with background color instead of uninitialized surface.
         let background_color = config.colors.primary.background;
         renderer.with_api(&config, &size_info, |api| {
             api.clear(background_color);
         });
 
-        // We should call `clear` when window is offscreen, so when `window.show()` happens it
-        // would be with background color instead of uninitialized surface.
+        #[cfg(not(any(target_os = "macos", windows)))]
+        let mut wayland_event_queue = None;
+
         #[cfg(not(any(target_os = "macos", windows)))]
         {
-            // On Wayland we can safely ignore this call, since the window isn't visible until you
-            // actually draw something into it.
+            // Setup and perform platform specific helpers
             if event_loop.is_x11() {
+                // On Wayland we can safely ignore this call, since the window isn't visible until
+                // you actually draw something into it.
                 window.swap_buffers()
+            } else {
+                // Initialize Wayland event queue, to handle various Wayland related routines.
```

Comment:
Do we really handle various Wayland related routines? Seems like we just handle frame callbacks.

If this is made because future additions might also need it, what kind could be integrated here?

Even if we wanted to keep this comment generic, to make sure things don't change in the future rendering this comment outdated without us noticing, I'd remove the word `various` since it's usually kind of useless.

```suggestion
                // Initialize event queue, to handle Wayland callbacks.
```

I've replaced `related routines` with `callbacks` since it's a bit more specific. But there might be better terms for it. If wayland-client names them 'routines' then that would be the better choice.